From: hololeap <hololeap@protonmail.com>
Signed-off-by: hololeap <hololeap@protonmail.com>

Add new USE flag: examples
This toggles building the example Jupyter kernels included in the package.

The executable names have been prepended with "jupyter-" to help avoid file
collisions.

diff -urN jupyter-0.9.0-orig/jupyter.cabal jupyter-0.9.0/jupyter.cabal
--- jupyter-0.9.0-orig/jupyter.cabal	2016-08-13 20:04:05.000000000 -0600
+++ jupyter-0.9.0/jupyter.cabal	2021-08-15 11:13:00.578668153 -0600
@@ -79,7 +79,13 @@
                  unordered-containers >= 0.2,
                  jupyter
 
-executable kernel-basic
+flag examples
+    default: False
+    description: Build example Jupyter kernel executables
+
+executable jupyter-kernel-basic
+    if !flag(examples)
+        buildable: False
     default-language:   Haskell2010
     hs-source-dirs:     examples/basic
     main-is:            Main.hs
@@ -87,7 +93,9 @@
                         text >= 1.2,
                         jupyter
 
-executable kernel-calculator
+executable jupyter-kernel-calculator
+    if !flag(examples)
+        buildable: False
     default-language:   Haskell2010
     hs-source-dirs:     examples/calculator
     main-is:            Main.hs
@@ -96,7 +104,9 @@
                         text >= 1.2,
                         jupyter
 
-executable kernel-stdin
+executable jupyter-kernel-stdin
+    if !flag(examples)
+        buildable: False
     default-language:   Haskell2010
     hs-source-dirs:     examples/stdin
     main-is:            Main.hs
@@ -104,7 +114,9 @@
                         text >= 1.2,
                         jupyter
 
-executable client-kernel-info
+executable jupyter-client-kernel-info
+    if !flag(examples)
+        buildable: False
     default-language:   Haskell2010
     hs-source-dirs:     examples/client-kernel-info
     main-is:            Main.hs
